File: PAT/BLevel/1012/1012.cpp
// 1012 数字分类
// auguralpha
// 2019-02-25

#include <cstdio>
#include <iostream>
using namespace std;
const int maxn = 1010;
int arr[maxn];
int cnt_a1 = 0;
int cnt_a2 = 0;
int cnt_a3 = 0;
int cnt_a4 = 0;
int cnt_a5 = 0;
int sum_a1 = 0;
int sum_a2 = 0;
double average_a4 = 0.0;
int max_a5 = 0;


int main() {
    int n;
    scanf("%d", &n);
    for(int i = 0; i < n; ++ i) {
        scanf("%d", &arr[i]);
    }
    for(int i = 0; i < n; ++ i) {
        if(arr[i] % 5 == 0 && arr[i] % 2 == 0) {
            sum_a1 += arr[i];
            ++ cnt_a1;
        }
        if(arr[i] % 5 == 1) {
            if(cnt_a2 % 2 == 0) {
                sum_a2 += arr[i];
            }
            else {
                sum_a2 -= arr[i];
            }
            ++ cnt_a2;
        }
        if(arr[i] % 5 == 2) {
            ++ cnt_a3;
        }
        if(arr[i] % 5 == 3) {
            average_a4 += arr[i];
            ++ cnt_a4;
        }
        if(arr[i] % 5 == 4) {
            max_a5 = max(max_a5, arr[i]);
            ++ cnt_a5;
        }
    }

    if(cnt_a1 != 0) {
        printf("%d ", sum_a1);
    }
    else {
        printf("N ");
    }
    if(cnt_a2 != 0) {
        printf("%d ", sum_a2);
    }
    else {
        printf("N ");
    }
    if(cnt_a3 != 0) {
        printf("%d ", cnt_a3);
    }
    else {
        printf("N ");
    }
    if(cnt_a4 != 0) {
        average_a4 = average_a4 / cnt_a4;
        printf("%.1f ", average_a4);
    }
    else {
        printf("N ");
    }
    if(cnt_a5 != 0) {
        printf("%d\n", max_a5);
    }
    else {
        printf("N\n");
    }
    return 0;
}

File: PAT/BLevel/1012/1012.go
// 1012 数字分类
// auguralpha
// 2020-03-25

package main

import (
	"fmt"
	"math"
)

func main() {
	var n int
	var a int
	var sumA1 int = 0
	var sumA2 int = 0
	var sumA4 float64 = 0.0
	var cntA2 int = 0
	var cntA3 int = 0
	var cntA4 int = 0
	var maxA5 int = 0
	_, _ = fmt.Scanf("%d", &n)
	for n != 0 {
		n --
		_, _ = fmt.Scanf("%d", &a)
		if a % 5 == 0 {
			if a % 2 ==  0 {
				sumA1 += a
			}
		} else if a % 5 == 1 {
			sumA2 += int(math.Pow(-1, float64(cntA2))) * a
			cntA2 += 1
		} else if a % 5 == 2 {
			cntA3 ++
		} else if a % 5 == 3 {
			sumA4 += float64(a)
			cntA4 ++
		} else {
			if a > maxA5 {
				maxA5 = a
			}
		}
	}

	if sumA1 != 0 {
		fmt.Printf("%d ", sumA1)
	} else {
		fmt.Print("N ")
	}

	if cntA2 != 0 {
		fmt.Printf("%d ", sumA2)
	} else {
		fmt.Print("N ")
	}

	if cntA3 != 0 {
		fmt.Printf("%d ", cntA3)
	} else {
		fmt.Print("N ")
	}

	if cntA4 != 0 {
		fmt.Printf("%.1f ", sumA4 / float64(cntA4))
	} else {
		fmt.Print("N ")
	}

	if maxA5 != 0 {
		fmt.Printf("%d\n", maxA5)
	} else {
		fmt.Print("N\n")
	}
}
